service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/ControlledAzureDiskResource.java

```diff
@@ -170,6 +170,9 @@ public void validate() {
       throw new MissingRequiredFieldException(
           "Missing required region field for ControlledAzureDisk.");
     }
+    if (!(getSize() > 0)) {
```

Comment:
Not really part of this PR, but found it was a bit odd we were allowing 0 when I was digging through the API's validation for testing

Reply:
Nit, this may be a bit simpler: 

```suggestion
    if (getSize() <= 0) {
```

Reply:
do we want to add a test (couldn't find one - but I may have missed it) to validate this is doing what we think?

Reply:
There are a few tests that validate the happy path here by creating valid disk resource that failed and needed updating from 0 -> 1 when this constraint was added. 

It seems the general practice for validation testing is that more complex validation that is contained in `Util` files (i.e. `AzureResourceValidationUtils` has a complex `validate` method for azure vms) is unit tested, but basic assertions are included in the constructor for these controlled resources (i.e. this  method in `ControlledAzureDiskResource`) and we don't test the unhappy path for these. Could have this assumption wrong, what is your recommendation @aherbst-broad ?

Reply:
We have not been diligent in getting unit tests around the build+validate code for our azure resources. A unit test would be good here. I imagine you could grab an instance of a disk from the `ControlledAzureResourceFixtures` class and make assertions to get the needed coverage, i.e,  `ControlledAzureResourceFixtures.getAzureDisk("foo", "bar", 0)`, etc.